src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -228,7 +228,7 @@ private async Task ProcessQueueAsync()
                         // Non mutating are fire-and-forget because they are by definition readonly. Any errors
                         // will be sent back to the client but we can still capture errors in queue processing
                         // via NFW, though these errors don't put us into a bad state as far as the rest of the queue goes.
-                        _ = ExecuteCallbackAsync(work, context, _cancelSource.Token).ReportNonFatalErrorAsync();
+                        _ = Task.Run(() => ExecuteCallbackAsync(work, context, _cancelSource.Token).ReportNonFatalErrorAsync(), _cancelSource.Token);
```

Comment:
I _think_ the reportnonfatalerrorasync should move to the Task.Run call, based on similar usages
https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/EditAndContinue/EditAndContinueSaveFileCommandHandler.cs,50

Reply:
Ah interesting ok!